huey_multitenant/core.py

```diff
@@ -166,8 +167,9 @@ def instance_is_active(self, instance):
 
     def consume_task(self):
         for idx, _instance in enumerate(self.instances):
-            # if self.instance_is_active(_instance):
-            #     continue
+            if _instance.is_in_maintenance_mode():
```

Comment:
para evitar el acople, no es mejor que se llame siempre y que sea la instancia la que decida si hace algo o si esta en mantenimiento?

Reply:
asi podemos evitar:
 def is_in_maintenance_mode(self):
queda solo en la instancia

Reply:
Vos decís cuando la instancia ya está procesando la tarea? El tema de ahí es que ya la desencoló de redis y no va a estar para cuando se levante de nuevo la aplicación.

Reply:
el multitenant no la levanta hace un peek.
la instancia no la va a quitar, va a decir estoy en mantenimiento, no quito nada. no?

Reply:
Tiene sentido. No revisé tan en detalle el código de consumir la tarea, ahora lo veo.

Lo único que me genera dudas es un tema de performance al hacer el peek. Potencialmente cada vez que lo hace está levantando una instancia de django

Reply:
lo que decis es cierto. esto es mas barato porq no haces spawn de un proceso que solo va a terminar.

Me preocupa esto:
# CAUTION: If you change the logic to decide if the system
        # is in maintenance mode or not, also change it in neo-assets

porque si mañana cambiamos la forma de procesar el modo mantenimiento, mientras actualizamos a todas las instancias, tenes 2 implementaciones y el multitenant tiene que conocer ambas. Se entiende?

Reply:
Sisi, se entiendo. Son los problemas que pueden surgir de tener 1 version de huey con N versiones de la aplicacion en simultaneo.

Mencionaba el tema de performance para tener cuidado con eso al mover el chequeo a la instancia, no para dejar las cosas como están.

Reply:
Una alternativa, validemos que tan cabeza es, sería:
- La instancia tiene un modulo conocido que se puede importar y valida si la misma está en mantenimiento.

Ventajas:
- Queda todo en la app y no se levantan procesos extras. Los detalles de implementacion del mismo quedan en la instancia.
Desventajas:
- Se me hace muy chancho